### PR TITLE
Do not copy empty strings in set_resource_type and set_interface_desc…

### PR DESCRIPTION
**Pull request by anttiylitokola**
…ription functions.

### CODE REVIEW

**Comment by anttiylitokola** [comment]
@TeroJaasko

File: source/m2mbase.cpp [hunk 1]
```diff
@@ -106,10 +106,14 @@ void M2MBase::set_interface_description(const char *desc)
...
     free(_sn_resource->dynamic_resource_params->static_resource_parameters->interface_description_ptr);
-    _sn_resource->dynamic_resource_params->static_resource_parameters->interface_description_ptr =
-            alloc_string_copy((uint8_t*) desc, strlen(desc));
-    _sn_resource->dynamic_resource_params->static_resource_parameters->interface_description_len =
-            strlen(desc);
+    _sn_resource->dynamic_resource_params->static_resource_parameters->interface_description_ptr = NULL;
+    const size_t len = strlen(desc);
+    if (len > 0 ) {
+        _sn_resource->dynamic_resource_params->static_resource_parameters->interface_description_ptr =
+                alloc_string_copy((uint8_t*) desc, strlen(desc));
```
Comment (TeroJaasko) [inline source/m2mbase.cpp]: You could be reusing the "len" variable instead of "strlen(desc)" to save some code space.

**Comment by TeroJaasko** [comment]
+1

**Comment by TeroJaasko** [comment]
Sorry, wrong button.

**Pull request merged**